apache2/msc_multipart.c

```diff
@@ -214,23 +214,42 @@ static int multipart_parse_content_disposition(modsec_rec *msr, char *c_d_value)
                 }
                 msr->multipart_filename = decoded_filename;
 
+               
+                if (msr->mpd->mpp->filenameext != NULL) {
+                    msr_log(msr, 4, "Multipart: Warning: Duplicate Content-Disposition filename*: %s",
+                     log_escape_nq(msr->mp, decoded_filename));
+                     return -17;
+                }
+
+                msr->mpd->mpp->filenameext = apr_pstrdup(msr->mp, decoded_filename);
+
+                // The `filename*` RCF 5987 encoded filename always overrides the `filename` parameter in content-disposition header.
+                msr->mpd->mpp->filename = msr->mpd->mpp->filenameext;
```

Comment:
instead of overwriting, is there anyway we could process both?

Reply:
As discussed, since we don't treat `filename` as a collection, we can't get rid of the override.